lib/jirawatch/cli/track.rb

```diff
@@ -6,55 +6,84 @@ class Track < Dry::CLI::Command
       include Jirawatch::CLI::AuthenticatedCommand
 
       argument :issue_key, required: true, desc: "Issue key that you want to track time for"
-      option :started_at, aliases: ["-t"], desc: "Specify when you started working on this issue with a HH:mm format"
+      option :time_started_at, aliases: ["-t"], desc: "Specify when you started working on this issue with a HH:mm format"
       option :worklog_message, aliases: ["-m"], desc: "Specify a work log message"
 
       def call(issue_key:, **options)
         @jira_client.Issue.find(issue_key) # Fails if issue doesn't exist
 
+        time_started_at = Time.parse(options.fetch(:time_started_at, Time.now.to_s))
+        time_paused_at = 0
+        time_restarted_at = time_started_at
+        partial_time_spent = 0
+
         puts "Logging time for #{issue_key}..."
-        started_at = Time.parse(options.fetch(:started_at, Time.now.to_s))
-        puts started_at
+        puts time_started_at
+        puts
+        puts "Press CTRL-P to pause/restart the timer"
+        puts "Press CTRL-C when you have finished to track the time"
+
+        reader = TTY::Reader.new
+        reader.on(:keyctrl_p) do
+          if time_paused_at == 0
+            time_paused_at = Time.now
+            partial_time_spent = partial_time_spent + (time_paused_at - time_restarted_at)
+
+            time_unit =
+                if (partial_time_spent / 60).floor == 1
```

Comment:
You don't need `floor` here, you're dividing it by an integer and the result will be an integer.

Also, remove parentheses! In Ruby, parentheses are ugly most of the time 😉

Reply:
The parentheses were necessary to apply the floor method.

However I used the floor because the subtraction between Times (row 30) returns seconds but with decimals (eg. `7.228868474`).

I investigated more and the solution is to convert the `Time.now` to an integer using the [`to_i` method](https://ruby-doc.org/core-2.7.1/Time.html#method-i-to_i). With the new commit all the floor are gone